dts/bindings/sensor/st,ism330dhcx-common.yaml

```diff
@@ -11,6 +11,26 @@ properties:
         The property value should ensure the flags properly describe
         the signal that is presented to the driver.
 
+    accel-odr:
+      type: int
+      required: false
+      default: 0
+      description:
+          Specify the default accelerometer output data rate expressed in samples per second (Hz).
+          Default is power-up configuration.
+      enum:
+        - 0  # Power-Down
```

Comment:
Same for this patch.

Even better since the enum values here do not correspond to the units users will be familiar with, you might want to have a look at the enhanced enum support that was added for Zephyr 2.5. This would make the DTS easier to read without sacrificing footprint.

So in YAML, something like this (with string type instead of int):

```yaml
      enum:
        - "power-down"
        - "12.5"
        - "26"
        - "52"
        - "104"
        - "208"
        - "416"
        - "833"
        - "1660"
        - "3330"
        - "6660"
```

And then in C:

```c
#define ACCEL_ODR_POWER_DOWN 0
#define ACCEL_ODR_12_5 1
#define ACCEL_ODR_26 2
#define ACCEL_ODR_52 3
#define ACCEL_ODR_104 4
#define ACCEL_ODR_208 5
#define ACCEL_ODR_416 6
#define ACCEL_ODR_833 7
#define ACCEL_ODR_1660 8
#define ACCEL_ODR_3330 9
#define ACCEL_ODR_6660 10

#define TO_ACCEL_ODR(inst) \
        ACCEL_ODR_ ## DT_ENUM_UPPER_TOKEN(DT_DRV_INST(inst), accel_odr)
```

So you can set the configuration field using `.accel_odr = TO_ACCEL_ODR(0),` in `ism330dhcx_config`.

(Not tested.)

Reply:
Mmmmh. I think I'll opt for just moving the enumeration info in the description part.